### PR TITLE
Make site translatable and start on Norwegian Bokmål (nb) translation.

### PR DESCRIPTION
**Pull request by petterreinholdtsen**
This fixes issue #4.

### CODE REVIEW

**Comment by petterreinholdtsen** [comment]
Note, this change is in conflict with other pull requests, as it changes several templates.  If this change is applied before the others, I can rebase the other change requests and add translation support to them as well.

File: fkbeta/fk/tests.py [hunk 1]
```diff
@@ -24,8 +25,8 @@ def test_video_listing(self):
     def test_video_detail(self):
         r = self.client.get('/video/1/')
         self.assertContains(
-            r, 'First broadcast: <i>Jan. 1, 2015, 10 a.m.</i>', count=1)
-        self.assertContains(r, '<h1>Video not available</h1>', count=1)
+            r, _('First broadcast:') + ' <i>', count=1)
```
Comment (odinho) [inline fkbeta/fk/tests.py]: Uhm. Should we run the tests with translation on? And I think you removed the useful bit of the test which is the rendering of the date :)

**Comment by petterreinholdtsen** [comment]
[Odin Hørthe Omdal]
> I think the messages should be in one place only. Collected instead of
> spread like now.

So do I, but I could not get it to work.  Suggestions?

> I don't think the tests should be influenced by the translation.

Any idea how to change the locale used when running tests?

> I think the static language code should be English, and it could be
> dynamically set to other languages (I guess we'll at maximum have
> `en`, `nb`, `nn`).

No idea how to do that.  Do you?

> Adding `gettext` should be its own commit (since we can add that to
> the project right away).

What do you mean by 'gettext'?  I am happy to split out separate
patches, if I understand what you suggest.

> Uhm. Should we run the tests with translation on? And I think you
> removed the useful bit of the test which is the rendering of the date
> :)

Did not know how to get the localized date string, so I removed it to
get the test working.  Assumed the test was checking the web page was
generated, not if the date was correct.  Suggestions welcome.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by odinho** [comment]
> [Odin Hørthe Omdal]
> So do I, but I could not get it to work. Suggestions?
Add to `LOCALE_PATHS`: https://docs.djangoproject.com/en/1.11/topics/i18n/translation/#how-django-discovers-translations

> Any idea how to change the locale used when running tests?
Not setting it to no-nb statically would do it. :) Or just having no-nb in settings/production.py (so dev and test won't get it).

> No idea how to do that. Do you?

Adding a middleware that checks what language the user prefers. Just like the middleware I just added to make API always return UTC dates.

There is also a `LocaleMiddleware` already: https://docs.djangoproject.com/en/2.1/ref/middleware/#module-django.middleware.locale

> What do you mean by 'gettext'? I am happy to split out separate patches, if I understand what you suggest.

I mean just adding all the `_` and `trans` thingies. But not actually adding all the message files to the project, and not doing any change. So it looks just as it does now, no run-time change. But everything is tagged and ready for us to later turn on translations, and create some.

**Comment by petterreinholdtsen** [comment]
I moved the first part, flagging strings as translatable, into pull request #114.

**Comment by toresbe** [comment]
Hey guys, thanks for the PR. I'm closing this because the issue was not resolved and the use of the Django website as a frontend is being aggressively phased out.

**Pull request closed without merging**